processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -271,19 +273,52 @@ public File merge(
     );
   }
 
+  private List<String> getLongestSharedDimOrder(List<IndexableAdapter> indexes)
+  {
+    int maxSize = 0;
+    Iterable<String> orderingCandidate = null;
+    for (IndexableAdapter index : indexes) {
+      int iterSize = Iterators.size(index.getDimensionNames().iterator());
+      if (iterSize > maxSize) {
+        maxSize = iterSize;
+        orderingCandidate = index.getDimensionNames();
+      }
+    }
+
+    if (orderingCandidate == null) {
+      return null;
+    }
+
+    for (IndexableAdapter index : indexes) {
+      Iterator<String> candidateIter = orderingCandidate.iterator();
+      for (String matchDim : index.getDimensionNames()) {
+        boolean matched = false;
+        while (candidateIter.hasNext()) {
```

Comment:
i see, if orderingCandidate is [A,B,C,D]
and say index.getDimensionNames is [B,D] this will pass but if it was [D,B] then it wouldn't .
ok, but still wondering why  [A,B,C,D] would be any less optimal than lexicographic when index had [D,B] dimension ordering.